teuthology/suite/run.py

```diff
@@ -129,8 +136,11 @@ def choose_os(self):
         os_type = self.args.distro
         os_version = self.args.distro_version
         if not (os_type and os_version):
-            os_ = util.get_distro_defaults(
-                self.args.distro, self.args.machine_type)[2]
+            try:
+                os_ = util.get_distro_defaults(
+                    self.args.distro, self.args.machine_type)[2]
+            except KeyError:
+                raise ScheduleFailError("distro {} doesn't exists".format(self.args.distro),'')
```

Comment:
```suggestion
                raise ScheduleFailError(f"distro {self.args.distro} doesn't exist")
```
(to fix the incorrect plural and also use f-strings for more concise syntax)

Reply:
thank you, resolved with the new commit.